### ISSUE TITLE
Sorting of non Journaled migrations

### ISSUE THREAD

**Issue by carl-berg** [mattiasnordqvist/Galactic-Waste-Management#38]
This issue is related to #37
If `netstandard2.0` is implemented, there's an interesting subtle problem in `InternalVersionComparer<T>.Compare` that needs to be solved (although, i think the problem can occur in `netcore` as well if there are more than one non journaled version being sorted). Long story short, either the compare method needs to handle null values or sorting must not be able to pass null values to comparer.

This code in `MigrationBase` uses this comparer
```csharp
if (version == null)
{
    orderedScripts = orderedScripts.ThenBy(x => x.Type.IsJournaled ? 
        ProjectSettings.MigrationVersioning.Version(x) : null, 
        ProjectSettings.MigrationVersioning.VersionComparer);
}
```
When this happens for `RunScripts(GetScripts(ScriptType.Initialize));` there is only one script (at least for me with my current setup) `CreateSchemaVersionJournal` that is being sorted here, so nothing is invoked on `InternalVersionComparer<T>.Compare`. This is however not true for `net472` where the same code makes one invocation with arguments `null, null`, even though there is only one element being sorted (super weird).

**Comment by carl-berg** [comment]
I have solved the issue like this in an internally packaged build of GWM targeting `netstandard2.0`:
```csharp
public int Compare(Version x, Version y)
{
    /// Non versioned journals should come before versioned
    if (x is null && y is null)
    {
        return 0;
    }
    else if (x is null)
    {
        return -1;
    }
    else if (y is null)
    {
        return 1;
    }

    return getVersionToCompare(x).CompareTo(getVersionToCompare(y));
}
```

**Comment by mattiasnordqvist** [comment]
did you manage to target netstandard without ripping out this https://www.nuget.org/packages/Microsoft.SqlServer.DacFx/150.4316.1-preview ?

**Comment by carl-berg** [comment]
It was fine after upgrading to the latest version

**Comment by mattiasnordqvist** [comment]
oh yeah thats right I have to fix this now too

**Comment by mattiasnordqvist** [comment]
fixed in v0.23

@all-contributors please add @carl-berg for bugs
